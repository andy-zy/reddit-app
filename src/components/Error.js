// @flow
import React, { Component } from 'react'
import {
  Text,
  View,
} from 'react-native'

import type { PropsBaseT } from '../domain/types'

import { layout, colors } from '../styles'

export default class Error extends Component<PropsBaseT> {

  static navigationOptions = {
    title: 'Error',
  }

  render() {
    const { navigation } = this.props
    const error: string = navigation.state.params.error || 'Something went wrong!'

    return (
      <View style={layout.vContainer}>
        <Text style={colors.error}>{error}</Text>
      </View>
    )
  }
}