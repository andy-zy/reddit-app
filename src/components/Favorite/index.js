import Favorite from './Favorite.container'

export default Favorite