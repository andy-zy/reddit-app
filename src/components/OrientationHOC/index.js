import OrientationHOC from './OrientationHOC'

export default OrientationHOC