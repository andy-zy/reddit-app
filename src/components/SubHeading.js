// @flow
import React from 'react'
import { Text } from 'react-native'

import { layout } from '../styles'

export default (props: Object) => <Text {...props} style={[layout.subHeading, props.style]}>{props.children}</Text>