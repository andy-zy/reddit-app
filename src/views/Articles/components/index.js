import Tabs from './Tabs'
import Search from './Search'
import ArticleList from './ArticleList'
import ListItem from './ListItem'

export {
  Tabs,
  Search,
  ArticleList,
  ListItem,
}