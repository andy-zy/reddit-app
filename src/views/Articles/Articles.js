// @flow
import React, { Component } from 'react'
import {
  View,
  Image,
  FlatList,
  ActivityIndicator,
  TouchableHighlight,
} from 'react-native'

import type { ArticleT } from '../../domain/types'
import type { PropsT, RenderT } from './types'

import { SubHeading } from '../../components'
import { layout } from '../../styles';
import styles from './styles'

export default class Articles extends Component<PropsT> {

  static navigationOptions = {
    title: 'ReactJS',
  }

  componentDidMount() {
    const { getArticlesByCategory } = this.props

    getArticlesByCategory(Articles.navigationOptions.title.toLocaleLowerCase())
  }

  renderItem = ({ item }: RenderT) => {
    const { orientation } = this.props
    const isPortrait: boolean = orientation === 'PORTRAIT'
    const imageSource = item.preview &&
      item.preview.images &&
      item.preview.images[0] &&
      item.preview.images[0].source;

    return (
      <TouchableHighlight onPress={this.handlePress(item)} style={styles.button}>
        <View style={[styles.item, styles.button, !isPortrait && styles.itemL]}>
          <View style={[styles.imageWrap, isPortrait ? styles.imageWrapP : styles.imageWrapL]} >
            {
              imageSource ? (
                <Image
                  source={{ uri: imageSource.url, height: imageSource.height, width: imageSource.width }}
                  style={styles.image}
                />
              ) : (
                <Image source={require('../../images/No_Image_Available.png')} style={styles.image} />
              )
            }
          </View>
          <SubHeading style={!isPortrait && styles.textL}>{item.title}</SubHeading>
        </View>
      </TouchableHighlight>
    )
  }

  renderSeparator = () => <View style={styles.separator} />

  keyExtractor = (item: ArticleT) => item.id

  handlePress = (item: ArticleT) => {
    const { setActiveArticle, navigation } = this.props

    return () => {
      setActiveArticle(item)
      navigation.navigate('Browser')
    }
  }

  render() {
    const {
      articles,
      isFetching,
      orientation,
    } = this.props

    const isPortrait: boolean = orientation === 'PORTRAIT'

    return (
      <View style={isPortrait ? layout.vContainer : layout.hContainer}>
        {
          isFetching ? <ActivityIndicator /> : (
            <FlatList
              data={articles}
              renderItem={this.renderItem}
              keyExtractor={this.keyExtractor}
              ItemSeparatorComponent={this.renderSeparator}
            />
          )
        }
      </View>
    )
  }
}
