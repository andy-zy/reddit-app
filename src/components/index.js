import { default as Favorite } from './Favorite'
import { default as OrientationHOC } from './OrientationHOC'

export {
  Favorite,
  OrientationHOC,
}