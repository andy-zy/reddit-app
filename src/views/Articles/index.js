import Articles from './Articles.container'

export default Articles