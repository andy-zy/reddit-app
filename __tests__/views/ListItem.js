import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { ListItem } from '../../src/views/Articles/components'

jest.mock('../../src/components/Favorite', () => {
  return require('../../src/components/Favorite/Favorite')
})

describe('ListItem', () => {
  let props

  beforeEach(() => {
    props = {
      item: { id: 'id1', title: 'title1', url: 'url1', preview: {} },
      orientation: 'PORTRAIT',
      onPress: jest.fn(),
    }
  })

  it('renders correctly', () => {
    const tree = renderer.create(<ListItem {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('item press should call onPress', () => {
    const component = shallow(<ListItem {...props} />)
    component.simulate('press')
    expect(props.onPress).toHaveBeenCalledWith(props.item)
  })
})