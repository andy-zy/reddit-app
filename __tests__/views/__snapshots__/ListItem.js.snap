// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "#FFF",
      "borderRadius": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "padding": 10,
        },
        Object {
          "borderColor": "#FFF",
          "borderRadius": 5,
        },
        false,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "marginBottom": 10,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/No_Image_Available.png",
          }
        }
        style={
          Object {
            "height": 160,
            "width": 160,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "justifyContent": "flex-start",
          },
          false,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/unmarked_favorite.png",
            }
          }
          style={
            Object {
              "height": 24,
              "width": 24,
            }
          }
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        title1
      </Text>
    </View>
  </View>
</View>
`;
